static/js/publisher/release/releasesController.js

```diff
@@ -17,14 +18,19 @@ import {
   getReleaseDataFromChannelMap
 } from "./releasesState";
 
-export default class ReleasesController extends Component {
+class ReleasesController extends Component {
   constructor(props) {
     super(props);
 
     // init channel data in revisions list
+    // TODO: should be done in reducers?
     const revisionsMap = getRevisionsMap(this.props.releasesData.revisions);
     initReleasesData(revisionsMap, this.props.releasesData.releases);
 
+    // init redux store
+    // TODO: should be done outside component as initial state?
+    this.props.updateRevisions(revisionsMap);
```

Comment:
Or even in `componentWillMount`?

Reply:
Ideally this step should be done before data gets into the component. Somewhere where 'initial state' is prepared. But also it's needed when the data is updated via `fetch`, so that's why I think about moving it to reducers in future.

So some new action "RELEASE_SUCCESS" will update the revisions list (and channels revision was released to). And it will be a state change performed by reducer.